project-generator/src/main/java/com/redhat/ipaas/project/converter/DefaultProjectGenerator.java

```diff
@@ -137,9 +136,11 @@ public DefaultProjectGenerator(ConnectorCatalog connectorCatalog, ProjectGenerat
     }
     */
 
-    private byte[] generateFlowYaml(GenerateProjectRequest request) throws JsonProcessingException {
+    private byte[] generateFlowYaml(Map<String, byte[]> contents, GenerateProjectRequest request) throws JsonProcessingException {
         Flow flow = new Flow();
+        int stepCounter[] = new int[]{0};
         request.getIntegration().getSteps().ifPresent(steps -> {
+            stepCounter[0]++;
```

Comment:
if done in the closure then it would reset or every step and not actually count up.